client/src/Try/Container.purs

```diff
@@ -438,34 +438,47 @@ renderCompilerErrors errors = do
     , renderPlaintext message
     ]
 
+type ShareButtonState =
+  { forkId :: Maybe H.ForkId
+  , showCopySucceeded :: Maybe Boolean
+  }
+
 shareButton :: forall q i o. H.Component q i o Aff
 shareButton =  H.mkComponent
-  { initialState: \_ -> 0
+  { initialState: \_ -> { forkId: Nothing, showCopySucceeded: Nothing }
   , render
   , eval: H.mkEval $ H.defaultEval
       { handleAction = handleAction
       }
   }
   where 
-  handleAction :: Unit -> H.HalogenM Int Unit () o Aff Unit
+  handleAction :: Unit -> H.HalogenM ShareButtonState Unit () o Aff Unit
   handleAction _ = do
-      url <- H.liftEffect $ window >>= location >>= href
-      H.liftAff $ makeAff \f -> do
-        runEffectFn3 copyToClipboard url (f (Right unit)) (f (Left $ Aff.error "Failed to copy to clipboard"))
-        mempty
-      H.modify_ (_ + 1)
+    H.gets _.forkId >>= traverse_ H.kill
+    url <- H.liftEffect $ window >>= location >>= href
+    copySucceeded <- H.liftAff $ makeAff \f -> do
+      runEffectFn3 copyToClipboard url (f (Right true)) (f (Right false))
+      mempty
+    H.modify_ _ { showCopySucceeded = Just copySucceeded }
+    forkId <- H.fork do
       H.liftAff $ delay (1_500.0 # Milliseconds) 
-      H.modify_ (_ - 1)
-  render :: Int -> H.ComponentHTML Unit () Aff
-  render n =
+      H.modify_ _ { showCopySucceeded = Nothing }
+    H.modify_ _ { forkId = Just forkId }
```

Comment:
Rather than modifying the state twice...
```purs
copySucceed <- ...
H.modify_ _ { showCopySucceeded = Just copySucceeded }
...
H.modify_ _ { forkId = Just forkId }
```
Why not do a single state update at the end?
```
copySucceed <- ...
forkId <- H.fork do ...
H.modify_ $ const { forkId: Just forkId, showCopySucceeded: Just copySucceeded }

Reply:
Updated here 1f4b17a83e0bd1dbe348b516897f4ccdd8c4eb3f